common/src/main/java/org/openmrs/analytics/ParquetUtil.java

```diff
@@ -80,32 +140,77 @@ synchronized private AvroConverter getConverter(String resourceType) {
 		return converterMap.get(resourceType);
 	}
 	
+	@VisibleForTesting
+	synchronized Path bestOutputFile(String resourceType) throws IOException {
+		java.nio.file.Path outputDir = fileSystem.getPath(getParquetPath(), resourceType);
+		Files.createDirectories(outputDir);
+		Stream<java.nio.file.Path> files = Files.list(outputDir);
+		Optional<Integer> maxIndex = files.map(f -> {
+			Matcher matcher = OUTPUT_PATTERN.matcher(f.toString());
+			if (matcher.matches()) {
+				return new Integer(matcher.group(1).replaceFirst("^0+(?!$)", ""));
+			}
+			return -1;
+		}).max(Integer::compareTo);
+		int bestIndex = maxIndex.map(integer -> integer + 1).orElse(0);
+		if (bestIndex > 99999) {
+			// TODO: Handle cases with more than 100K files in a directory, e.g., by creating an extra
+			// layer in the directory hierarchy.
+			throw new IOException(
+			        "It is not wise to create more than 100K files in a directory, please adjust your configuration params!");
```

Comment:
I think I am missing something, where is this configured?

Reply:
Ooh never mind, I now get it!

Reply:
Just to make sure: This is coming from the `output-streaming-DDDDD` pattern, i.e., five digits as the suffix.